electron-utils/oauth.ts

```diff
@@ -2,8 +2,8 @@ import { BrowserWindow, shell } from 'electron';
 import { v4 as uuid } from 'uuid';
 
 const nodeUrl = require('url');
-const fetch = require('node-fetch');
 const Logger = require('electron-log');
+const fetch = (...args) => import('node-fetch').then(({ default: fetch }) => fetch(args[0], ...args.slice(1)));
```

Comment:
Can I check why ```args[0], ...args.slice(1)``` has to be used instead of just ```...args```?

Reply:
I think TypeScript will complain since fetch expects at least one argument. Maybe a better way is to do the following:
```suggestion
const fetch = (first, ...args) => import('node-fetch').then(({ default: fetch }) => fetch(first, ...args);
```